modulus/launch/utils/checkpoint.py

```diff
@@ -185,6 +185,7 @@ def save_checkpoint(
     scheduler: Union[scheduler, None] = None,
     scaler: Union[scaler, None] = None,
     epoch: Union[int, None] = None,
+    metadata: Union[Dict[str, Any], None] = None,
```

Comment:
Using `Optional` would be IMO more clear and shorter:
```py
 metadata: Optional[dict[str, Any]] = None
```
Same in other places.

Reply:
Fixed it